network/servicenetwork/watermill.go

```diff
@@ -88,17 +88,15 @@ func (n *ServiceNetwork) SendMessageHandler(msg *message.Message) error {
 }
 
 func (n *ServiceNetwork) sendMessage(ctx context.Context, msg *message.Message) error {
-	meta := payload.Meta{}
-	err := meta.Unmarshal(msg.Payload)
-	if err != nil {
-		return errors.Wrap(err, "failed to unwrap message")
+	receiver := msg.Metadata.Get(meta.Receiver)
+	if receiver == "" {
+		return errors.New("failed to send message: Receiver in message metadata is not set")
 	}
-	if meta.Receiver.IsEmpty() {
-		return errors.New("failed to send message: Receiver in meta message not set")
+	node, err := insolar.NewReferenceFromString(receiver)
```

Comment:
probably you still need here `node.Receiver.IsEmpty()`

Reply:
yeap, fixed